zerver/actions/create_user.py

```diff
@@ -547,6 +547,26 @@ def do_reactivate_user(user_profile: UserProfile, *, acting_user: Optional[UserP
                 }
             ).decode(),
         )
+
+        bot_owner_changed = False
+        if (
+            user_profile.is_bot
+            and user_profile.bot_owner is not None
+            and not user_profile.bot_owner.is_active
+            and acting_user is not None
+        ):
+            previous_owner = user_profile.bot_owner
+            user_profile.bot_owner = acting_user
+            user_profile.save()  # Can't use update_fields because of how the foreign key works.
+            RealmAuditLog.objects.create(
+                realm=user_profile.realm,
+                acting_user=acting_user,
+                modified_user=user_profile,
+                event_type=RealmAuditLog.USER_BOT_OWNER_CHANGED,
+                event_time=event_time,
+            )
+            bot_owner_changed = True
```

Comment:
Have not directly used `do_change_bot_owner` here to maintain the order of events and also to make database changes first and then sending events.